src/app/modules/shared/components/supplier/edit-supplier/edit-supplier.component.ts

```diff
@@ -75,18 +72,33 @@ export class EditSupplierComponent implements OnInit {
     this.closeDiv.nativeElement.click();
     const form = this.supplierForm.value;
 
+    let currentCurrency = null;
+    this.currencies.forEach(element => {
+      if(element.id === parseInt(form.currency)){
+        currentCurrency = element;
+      }
+      
+    });
+
+    let currentCountry = null;
+    this.countries.forEach(element => {
```

Comment:
currentCountry = this.countries[this.countries.findIndex(element => element.id === parseInt(form.country))];

Reply:
I think it will be better then cycle and if operator